src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2016,14 +2015,14 @@ public override BoundNode VisitCall(BoundCall node)
                 if (!receiverType.IsNull)
                 {
                     // Update method based on inferred receiver type.
-                    method = (MethodSymbol)AsMemberOfResultType(receiverType, method);
+                    method = (MethodSymbol)AsMemberOfResultType(receiverType, method)!;
                 }
 
                 ImmutableArray<RefKind> refKindsOpt = node.ArgumentRefKindsOpt;
                 (ImmutableArray<BoundExpression> arguments, ImmutableArray<Conversion> conversions) = RemoveArgumentConversions(node.Arguments, refKindsOpt);
                 ImmutableArray<int> argsToParamsOpt = node.ArgsToParamsOpt;
 
-                method = VisitArguments(node, arguments, refKindsOpt, method.Parameters, argsToParamsOpt, node.Expanded, node.InvokedAsExtensionMethod, conversions, method);
+                method = VisitArguments(node, arguments, refKindsOpt, method.Parameters, argsToParamsOpt, node.Expanded, node.InvokedAsExtensionMethod, conversions, method)!;
```

Comment:
>method = VisitArguments(node, arguments, refKindsOpt, method.Parameters, argsToParamsOpt, node.Expanded, node.InvokedAsExtensionMethod, conversions, method)!; [](start = 16, length = 158)

It feels like we can benefit from an attribute saying the result is not nullable if some parameter is not nullable.

Reply:
Yes, once such an attribute is supported.

---
In reply to: [241506605](https://github.com/dotnet/roslyn/pull/31719#discussion_r241506605) [](ancestors = 241506605)